h/storage.h

```diff
@@ -0,0 +1,47 @@
+#ifndef STORAGE_H
+#define STORAGE_H
+
```

Comment:
I think we should start documenting the APIs. For example, see https://github.com/zowe/zowe-common-c/blob/master/h/cellpool.h. I've started using doxygen doc in new files.